src/_pytest/outcomes.py

```diff
@@ -116,7 +132,9 @@ def skip(msg: str = "", *, allow_module_level: bool = False) -> "NoReturn":
 skip.Exception = Skipped  # type: ignore
 
 
-def fail(msg: str = "", pytrace: bool = True) -> "NoReturn":
+def fail(
+    msg: str = "", pytrace: bool = True, short_msg: Optional[str] = None
```

Comment:
We should make this a keyword only argument because it is a public API.

```suggestion
    msg: str = "", pytrace: bool = True, *, short_msg: Optional[str] = None
```